### ISSUE TITLE
Assert in mvcc `test_stmt->txn->psn == 0`

### ISSUE THREAD

**Issue by Serpentian** [tarantool/tarantool#7977]
I got the following output when running `storage-luatest/bucket_triggers_test.lua`:
```text
[004] replica_1_b | tarantool: ./src/box/memtx_tx.c:2464: memtx_tx_history_prepare_insert_stmt: Assertion `test_stmt->txn->psn == 0' failed.
[004] not ok 16	storage.memtx_use_mvcc_engine:true.test_bucket_space_reject_bad_replace_refs
[004] #   Peer closed
[004] #   stack traceback:
[004] #   	builtin/box/net_box.lua:817: in function '_request'
[004] #   	builtin/box/net_box.lua:878: in function 'eval'
[004] #   	.../Programming/Work/vshard/test/luatest_helpers/server.lua:327: in function 'wait_vclock_of'
[004] #   	...ork/vshard/test/storage-luatest/bucket_triggers_test.lua:385: in function 'storage.memtx_use_mvcc_engine:true.test_bucket_space_reject_bad_replace_refs'
[004] #   	...
[004] #   	[C]: in function 'xpcall'
```

Steps to reproduce:
1. Take branch https://github.com/Serpentian/vshard/tree/mvcc-crash2
2. Run `python test-run.py -j10 $(yes storage-luatest/bucket_triggers_test.lua | head -10)`
3. Sooner or later it fails on one of the following tests:
    * `test_bucket_space_commit_trigger_active`;
    * `test_group.test_bucket_space_commit_trigger_receiving`;
    * `test_bucket_space_reject_bad_replace_refs`;

Logs and xlogs are also attached to the ticket:
[004_storage-luatest.tar.gz](https://github.com/tarantool/tarantool/files/10097189/004_storage-luatest.tar.gz)

**Comment by Gerold103** [comment]
Might be a duplicate of https://github.com/tarantool/tarantool/issues/7487.

**Comment by Serpentian** [comment]
Closed as a duplicate of #7487
